internal/infrastructure/kubernetes/deployment.go

```diff
@@ -248,22 +208,31 @@ func expectedContainers(infra *ir.Infra) ([]corev1.Container, error) {
 }
 
 // createDeployment creates a Deployment in the kube api server based on the provided
-// infra, if it doesn't exist.
-func (i *Infra) createDeployment(ctx context.Context, infra *ir.Infra) (*appsv1.Deployment, error) {
-	expected, err := i.expectedDeployment(infra)
+// infra, if it doesn't exist and updates it if it does.
+func (i *Infra) createOrUpdateDeployment(ctx context.Context, infra *ir.Infra) error {
+	deploy, err := i.expectedDeployment(infra)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	if err := i.Client.Create(ctx, expected); err != nil {
+	if err := i.Client.Create(ctx, deploy); err != nil {
 		if kerrors.IsAlreadyExists(err) {
-			return expected, nil
+			// Update deployment if it exists.
+			if err := i.Client.Update(ctx, deploy); err != nil {
+				return fmt.Errorf("failed to update deployment %s/%s: %w",
+					deploy.Namespace, deploy.Name, err)
```

Comment:
IMO we should compare expected with actual, e.g. what exists in the controller-runtime cache, and only update if managed fields, e.g. `spec`, change. I'm fine proceeding with the approach in this PR for v0.2.0 as long as a tracking issue is created for doing the comparison.

Reply:
addressed your feedback in this PR